backfill.sql

```diff
@@ -167,12 +169,32 @@ BEGIN
     
     --And our time dimension, which is always the first dimension
     SELECT d.* INTO STRICT dimension_row FROM _timescaledb_catalog.dimension d WHERE hypertable_id = hypertable_row.id ORDER BY id LIMIT 1 ;
-    
-    -- Push the compression job out for some period of time so we don't end up compressing a decompressed chunk 
-    -- Don't disable completely because at least then if we fail and fail to move it back things won't get completely weird
-    SELECT move_compression_job(hypertable_row.id, hypertable_row.schema_name, hypertable_row.table_name, now() + compression_job_push_interval) INTO old_compression_job_time;
+
     COMMIT;
 
+    -- Push the compression job out for some period of time so we don't end up compressing a decompressed chunk
+    -- Don't disable completely because at least then if we fail and fail to move it back things won't get completely weird
+    LOOP
+        SELECT
+            move_compression_job(
+                hypertable_row.id,
+                hypertable_row.schema_name,
+                hypertable_row.table_name,
+                now() + compression_job_push_interval
+            )
+        INTO old_compression_job_time;
+        IF old_compression_job_time = '-infinity' :: timestamptz THEN
+            ROLLBACK;
+            RAISE NOTICE 'Compression job already running, sleeping...';
+            PERFORM pg_sleep(10);
+        ELSE
+            COMMIT;
+            RAISE NOTICE 'Compression job not already running, proceeding as normal...';
+            EXIT;
+        END IF;
+    END LOOP;
```

Comment:
If there are consecutive failures and you configure the `max_retries` option for the job the `next_start` will return NULL, so I guess you're not dealing with this case here. I mean the flow will continue to proceeding as normal... is this the expected behavior? Wondering if we'll not enter in a infinity loop here.

Reply:
Not sure, to be honest - it's a long time since we looked at this.

All I know is that we've used this successfully and not hit an infinite loop 🤷🏻